apps/server/src/modules/school/service/school.service.ts

```diff
@@ -31,7 +31,12 @@ export class SchoolService {
 	): Promise<OauthMigrationDto> {
 		const schoolDo: SchoolDO = await this.schoolRepo.findById(schoolId);
 		if (oauthMigrationPossible != null) {
-			schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
+			if (!schoolDo.oauthMigrationFinished && !schoolDo.oauthMigrationPossible) {
+				schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
+				schoolDo.oauthMigrationStart = schoolDo.oauthMigrationPossible;
+			} else {
+				schoolDo.oauthMigrationPossible = oauthMigrationPossible ? new Date() : undefined;
+			}
 		}
 		if (oauthMigrationMandatory != null) {
```

Comment:
If the checks are save?  
oauthMigrationMandatory?  ... undefined, true, false
oauthMigrationMandatory != null  ...not fully connected to the possible param inputs

Reply:
yeah, this whole 1 endpoint handles all has gotten out of hand. We will refactor this in the next few sprints by breaking up this 1 endpoint into 3 with clear namings and use cases. This should lower the complexity and add greatly to the readability of the migration logic...